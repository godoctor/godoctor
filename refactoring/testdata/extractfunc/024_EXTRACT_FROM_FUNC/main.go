//<<<<<extract,11,2,15,2,Foo,pass
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func f(n int) {
	for i := 0; i < 5; i++ {
		fmt.Println(n, ":", i)
		amt := time.Duration(rand.Intn(250))
		time.Sleep(time.Millisecond * amt)
	}
	fmt.Println("Statement after for loop", n)
}

func main() {
	for i := 0; i < 5; i++ {
		go f(i)
	}
	var input string
	fmt.Scanln(&input)
}
