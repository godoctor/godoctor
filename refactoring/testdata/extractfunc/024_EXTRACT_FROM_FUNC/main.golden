//<<<<<extract,11,2,15,2,Foo,pass
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func f(n int) {
	Foo(n)
	fmt.Println("Statement after for loop", n)
}

func main() {
	for i := 0; i < 5; i++ {
		go f(i)
	}
	var input string
	fmt.Scanln(&input)
}
func Foo(n int) {
	for i := 0; i < 5; i++ {
		fmt.Println(n, ":", i)
		amt := time.Duration(rand.Intn(250))
		time.Sleep(time.Millisecond * amt)
	}
}
